### DISCUSSION TITLE
Error Handling Issue on AI Response with Multiple Models

### DISCUSSION THREAD

**Discussion by PedroGruvhagen** [danny-avila/LibreChat#2906] [Troubleshooting] [answered]
### What happened?

Our system is encountering an error when attempting to send messages through the AI response handling mechanism. The issue manifests as an HTTP 404 error, and the error message indicates a misconfiguration regarding model endpoints. While claude 3 and assistant functionality are unaffected, there is a failure with groq, OpenAI, and Azure OpenAI models.

 2024-05-29 14:32:29 error: [handleAbortError] AI response error; aborting request: Failed to send message. HTTP 404 - {
LibreChat         |   "error": {
LibreChat         |     "message": "This is a chat model and not supported in the v1/completions endpoint. Did you mean to use v1/chat/completions?",
LibreChat         |     "type": "invalid_request_error",
LibreChat         |     "param": "model",
LibreChat         |     "code": null
LibreChat         |   }
LibreChat         | }

### Steps to Reproduce

1: Initiate a message request using the groq, OpenAI, or Azure OpenAI integration.
2: Observe the response handling error and subsequent request abortion.
3: Inspect the logs for the detailed HTTP 404 error.

### What browsers are you seeing the problem on?

Chrome, Safari, Microsoft Edge, Mobile (iOS)

### Relevant log output

```shell
2024-05-29 14:32:29 error: [handleAbortError] AI response error; aborting request: Failed to send message. HTTP 404 - {
LibreChat         |   "error": {
LibreChat         |     "message": "This is a chat model and not supported in the v1/completions endpoint. Did you mean to use v1/chat/completions?",
LibreChat         |     "type": "invalid_request_error",
LibreChat         |     "param": "model",
LibreChat         |     "code": null
LibreChat         |   }
LibreChat         | }
```

### Screenshots

![image](https://github.com/danny-avila/LibreChat/assets/129173525/b661d7aa-c8e3-4094-aa9d-6c2011cdf4d6)
![image](https://github.com/danny-avila/LibreChat/assets/129173525/fe88aa02-d640-442c-b594-fce60be822a7)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I can't reproduce this: 

![image](https://github.com/danny-avila/LibreChat/assets/110412045/516e35ad-14a1-4720-8b4d-fdd4dec81a1e)

Please use `git log` and output the latest commit hash you see. 

Please share the output from the following command (linux based):

`docker inspect --format='{{.Config.Image}}' LibreChat`

**Reply by PedroGruvhagen** [reply]
git log:
_

> PS E:\LibreChat> git log
> commit e2cb2905e7ee27d52f491c9ab734bc196c8a3b06 (HEAD -> main, origin/main, origin/HEAD)
> Author: Danny Avila <danny@librechat.ai>
> Date:   Wed May 29 10:14:30 2024 -0400
> 
>     🖱️ feat: Minor Accessibility Changes (#2903)
> 
>     * feat: focus rings for dialog buttons
> 
>     * chore: temp: soft removal of removeFocusOutlines
> 
>     * feat: allow tabbing of endpoint menu
> 
> commit 3f600f0d3fa349939dbcfc9120b342a9b8e17377
> Author: Danny Avila <danny@librechat.ai>
> Date:   Wed May 29 09:15:05 2024 -0400
> 
>     ⬇️ fix: JSON LibreChat Imports (#2897)
> 
>     * chore: remove unused code
> 
>     * refactor: Update NewChatButtonIcon component to use JSX syntax
> 
>     The NewChatButtonIcon component in the Nav folder has been updated to use JSX syntax instead of calling the Icon function directly. This change improves code readability and maintainability.
> 
>     * remove use memo
> 
>     * refactor: allow passing `select` to messages db query
> 
>     * fix: initial fix for non-recursive messages
> 
>     * ci: first pass, importers test rewrite
> 
>     * fix(groupConversationsByDate): handle edge case of conversation.updatedAt being null
> 
>     * fix: correctly handle non-recursive uploads

_
    
    
    
    
    
    
docker inspect --format='{{.Config.Image}}' LibreChat:


_```

> PS E:\LibreChat> docker inspect --format='{{.Config.Image}}' LibreChat
> ghcr.io/danny-avila/librechat-dev:latest
> PS E:\LibreChat>
> ![image](https://github.com/danny-avila/LibreChat/assets/129173525/268ab45f-aaa1-49ad-af27-b915f3fe4aae)

```_

**Reply by PedroGruvhagen** [reply]
Many thanks for the super fast help and response!

**Reply by danny-avila** [reply]
@PedroGruvhagen can you try the steps highlighted on this page?

https://www.librechat.ai/docs/local/docker#update-librechat

After doing so, can you try to reproduce the error, and then share output from the latest debug logs, found in `./logs` of the project directory? Only the parts relevant to the request that led to the error.

Are you using a reverse proxy for any of the services? i.e. `OPENAI_REVERSE_PROXY` in .env file.

That could be one of the issues. Or using `OPENAI_FORCE_PROMPT` env. variable

**Reply by PedroGruvhagen** [reply]
Hey, many thanks for the amazing help; OPENAI_FORCE_PROMPT was uncommented; as soon as I commented that line out, everything started working.

What does that line have to do with Groq??? 

Anyways, the case is closed! many thanks!

**Reply by danny-avila** [reply]
It’s a legacy option, and groq/custom endpoints are being handled by the same class used for OpenAI

**Comment by PedroGruvhagen** [comment, marked as answer]
Using OPENAI_FORCE_PROMPT env. variable caused the issue
